compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -317,7 +319,7 @@ object Completion:
    *  For the results of all `xyzCompletions` methods term names and type names are always treated as different keys in the same map
    *  and they never conflict with each other.
    */
-  class Completer(val mode: Mode, val prefix: String, pos: SourcePosition):
+  class Completer(val mode: Mode, val prefix: String, pos: SourcePosition, prefixFilter: Name => Boolean):
```

Comment:
```suggestion
  class Completer(val mode: Mode, pos: SourcePosition, matches: Name => Boolean):
```
Do you even need the prefix anymore? Also update scaladoc.